test/hotspot/jtreg/compiler/loopopts/superword/TestMulAddS2I.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8310886
+ * @requires os.arch == "x86_64" | os.arch == "aarch64"
+ * @summary Test MulAddS2I vectorization.
+ * @library /test/lib /
+ * @run driver compiler.loopopts.superword.TestMulAddS2I
+ */
+
+package compiler.loopopts.superword;
+
+import compiler.lib.ir_framework.*;
+import jdk.test.lib.Asserts;
+import jdk.test.lib.Platform;
+
+public class TestMulAddS2I {
+    static final int RANGE = 1024;
+    static final int ITER  = RANGE/2 - 1;
+
+    static short[] sArr1 = new short[RANGE];
+    static short[] sArr2 = new short[RANGE];
+    static final int[] GOLDEN;
+
+    static {
+        for (int i = 0; i < RANGE; i++) {
+            sArr1[i] = (short)(i + (1000 * AbstractInfo.getRandom().nextDouble()));
+            sArr2[i] = (short)(RANGE - i + (2000 * AbstractInfo.getRandom().nextDouble()));
```

Comment:
Did you want to change this to `(short)AbstractInfo.getRandom().nextInt();`

Reply:
Good idea, I will update this.